src/Microsoft.ML.ImageAnalytics/VectorToImageTransform.cs

```diff
@@ -226,195 +533,63 @@ internal void Save(ModelSaveContext ctx)
+        ///<summary>
+        /// Extract pixels values from image and produce array of values.
+        ///</summary>
+        /// <param name="env">The host environment.</param>
+        /// <param name="columns">Describes the parameters of pixel extraction for each column pair.</param>
+        internal VectorToImageConvertingEstimator(IHostEnvironment env, params ColumnInfo[] columns)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(nameof(VectorToImageConvertingEstimator)), new VectorToImageConvertingTransformer(env, columns))
         {
-            Host.AssertNonEmpty(Infos);
-            Host.Assert(Infos.Length == Utils.Size(args.Columns));
-
-            _exes = new ColInfoEx[Infos.Length];
-            _types = new ImageType[Infos.Length];
-            for (int i = 0; i < _exes.Length; i++)
-            {
-                var item = args.Columns[i];
-                _exes[i] = new ColInfoEx(item, args);
-                _types[i] = new ImageType(_exes[i].Height, _exes[i].Width);
-            }
-            Metadata.Seal();
         }
 
-        private VectorToImageTransform(IHost host, ModelLoadContext ctx, IDataView input)
-            : base(host, ctx, input, t => t is VectorType ? null : "Expected VectorType type")
+        /// <summary>
+        /// Returns the <see cref="SchemaShape"/> of the schema which will be produced by the transformer.
+        /// Used for schema propagation and verification in a pipeline.
+        /// </summary>
+        public override SchemaShape GetOutputSchema(SchemaShape inputSchema)
         {
-            Host.AssertValue(ctx);
-
-            // *** Binary format ***
-            // <prefix handled in static Create method>
-            // <base>
-            // foreach added column
-            //   ColInfoEx
-            Host.AssertNonEmpty(Infos);
-            _exes = new ColInfoEx[Infos.Length];
-            _types = new ImageType[Infos.Length];
-            for (int i = 0; i < _exes.Length; i++)
+            Host.CheckValue(inputSchema, nameof(inputSchema));
+            var result = inputSchema.ToDictionary(x => x.Name);
+            foreach (var colInfo in Transformer.Columns)
             {
-                _exes[i] = new ColInfoEx(ctx);
-                _types[i] = new ImageType(_exes[i].Height, _exes[i].Width);
-            }
-            Metadata.Seal();
-        }
+                if (!inputSchema.TryFindColumn(colInfo.InputColumnName, out var col))
+                    throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", colInfo.InputColumnName);
+                if (col.Kind != SchemaShape.Column.VectorKind.Vector || (col.ItemType != NumberDataViewType.Single && col.ItemType != NumberDataViewType.Double && col.ItemType != NumberDataViewType.Byte))
+                    throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", colInfo.InputColumnName, new ImageType().ToString(), col.GetTypeString());
```

Comment:
>new ImageType().ToString() [](start = 107, length = 26)

This is not correct. You're saying the expected input type is an image type, yet, you are testing whether it is a known-size vector of float, double, or byte above.